proxy.go

```diff
@@ -33,26 +35,22 @@ var errProxyTimeout = errors.New("all peers timed out")
 //     case the code just waits for one to finish. If the total 'proxy_timeout'
 //     is hit at any point, the method returns immediately with an error and
 //     cancels any running requests.
-func (vs *version) proxy(w http.ResponseWriter, r *http.Request, peers []string) ([]byte, error) {
+func (vs *version) proxy(peers []string) (*http.Response, string, error) {
```

Comment:
I might label these return values for documentation purposes. I have no idea what the `string` return is, for instance